### PR TITLE
Update zig-cc and zigbuild versions.

### LINKED ISSUE #1109: Update Zig to 0.10.0
Currently, cross uses zig `0.9.1`, however, `0.10.0` was recently [released](https://ziglang.org/download/#release-0.10.0). We should upgrade the images to the latest version.

### PR DESCRIPTION
**Pull request by Alexhuszagh**
Closes #1109.

### CODE REVIEW

File: .changes/880-1111.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "description": "added a zig-based image (v0.10.0), allowing multiple targets to be built from the same image, using cargo-zigbuild (v0.14.1).",
```
Comment (Alexhuszagh) [inline .changes/880-1111.json:2]: I've added the specific versions to the description so these can be modified until the actual 0.3.0 release, and just link every PR associated.

File: ci/test-zig-image.sh [hunk 2]
```diff
@@ -20,7 +20,7 @@ TARGETS=(
     "i586-unknown-linux-musl"
 )
 
-# on CI, it sets `CROSS_TARGET_ZIG_IMAGE` rather than `CROSS_TARGET_ZIG_IMAGE`
+# on CI, it sets `CROSS_TARGET_ZIG_IMAGE` rather than `CROSS_BUILD_ZIG_IMAGE`
```
Comment (Alexhuszagh) [inline ci/test-zig-image.sh:23]: Whoops.

**Comment by Alexhuszagh** [comment]
bors try --none

**Comment by Alexhuszagh** [comment]
bors try --target zig

**Comment by Alexhuszagh** [comment]
bors try-
bors try --target zig

**Review by Emilgardis — APPROVED** [review]

**Comment by Alexhuszagh** [comment]
bors r=Emilgardis

**Pull request merged**